### DISCUSSION TITLE
Using Cypress for E2E tests

### DISCUSSION THREAD

**Discussion by harry-73** [JorgenVatle/meteor-vite#125] [Q&A] [answered]
Hi,

I am trying to setup Cypress working with this package on Github actions.

on Github, when running the tests, the `node_modules/.vite/dep/` folder is empty, so `vite`  optimized dependencies:

```
 => Client modified -- refreshing
I20240223-18:44:22.903(1)? 6:44:22 PM [vite] ✨ new dependencies optimized: vuetify/lib/components/VSkeletonLoader/index.mjs, vuetify/lib/components/transitions/index.mjs, vue-chartjs, chart.js, chartjs-adapter-moment
I20240223-18:44:22.903(1)? 6:44:22 PM [vite] ✨ optimized dependencies changed. reloading
```
and sometimes the Vite server restarts:

`I20240223-18:56:52.300(1)? ⚡   Starting Vite server..`

In this case, the tests failed.

I reproduce the issue, locally:

- I delete the `node_modules/.vite/dep/` folder
- I ran the tests ==> It fails
- the  `node_modules/.vite/dep/` folder has the _optimized_ files
- I run the tests ==> it is OK.

How can we avoid that?

**Comment by JorgenVatle** [comment, marked as answer]
Vite will optimize dependencies as they are being requested. It doesn't know which dependencies need to be optimized until you load the app for the first time. Probably because we're not using the standard Vite index.html file. So there should be some changes we can make to fix this for you with the next release.

For now, you can manually point Vite to your client entrypoint using [`optimizeDeps.entries`](https://v4.vitejs.dev/config/dep-optimization-options.html#optimizedeps-entries)

Let me know how it goes. 🤞

**Reply by JorgenVatle** [reply]
This should be fixed in `meteor-vite@1.10.1` 👍
